crates/biome_graphql_parser/src/parser/definitions/union.rs

```diff
@@ -39,6 +40,29 @@ pub(crate) fn parse_union_type_definition(p: &mut GraphqlParser) -> ParsedSyntax
     Present(m.complete(p, GRAPHQL_UNION_TYPE_DEFINITION))
 }
 
+/// Must only be called if the next 2 token is `extend` and `interface`, otherwise it will panic.
+#[inline]
+pub(super) fn parse_union_type_extension(p: &mut GraphqlParser) -> ParsedSyntax {
+    let m = p.start();
+
+    p.bump(T![extend]);
+    p.bump(T![union]);
+
+    parse_name(p).or_add_diagnostic(p, expected_name);
+
+    let pos = p.source().position();
+    DirectiveList.parse_list(p);
```

Comment:
nit: you can use something like this:

```
    let list = DirectiveList.parse_list(p);
    let list_range = list.range(p);

    if list_range.is_empty() {
```

Reply:
Oh, I never knew about this
Thanks a lot 😁